Jellyfin.Api/Controllers/FilterController.cs

```diff
@@ -71,11 +71,19 @@ public FilterController(ILibraryManager libraryManager, IUserManager userManager
                 parentItem = null;
             }
 
-            var item = string.IsNullOrEmpty(parentId)
-                ? user == null
-                    ? _libraryManager.RootFolder
-                    : _libraryManager.GetUserRootFolder()
-                : parentItem;
+            BaseItem? item;
+            if (parentId.HasValue)
```

Comment:
Lots of variations of this piece of code 😆 It would be very nice if you can change them all to use GetParentItem

Reply:
I think removing line 57-59 and 75-86 and doing this simplifies it a bit. It also has the benefit of _not_ fetching the parent item if it's just going to be discarded.
```c#
            BaseItem? item;
            if (includeItemTypes.Length == 1
                && (string.Equals(includeItemTypes[0], nameof(BoxSet), StringComparison.OrdinalIgnoreCase)
                    || string.Equals(includeItemTypes[0], nameof(Playlist), StringComparison.OrdinalIgnoreCase)
                    || string.Equals(includeItemTypes[0], nameof(Trailer), StringComparison.OrdinalIgnoreCase)
                    || string.Equals(includeItemTypes[0], "Program", StringComparison.OrdinalIgnoreCase)))
            {
                item= null;
            }
            else
            {
                item = _libraryManager.GetParentItem(parentId, user?.Id);
            }
```